### PR TITLE
[HAC-341] update README, add HAC nav and CRD

### PR DESCRIPTION
**Pull request by BlakeHolifield**

### CODE REVIEW

**Comment by BlakeHolifield** [comment]
hacCore

Shows navigation working based on the CRDs

File: controllers/reconcile.go [hunk 1]
```diff
@@ -401,7 +404,10 @@ func createConfigConfigMap(ctx context.Context, pClient client.Client, frontend
 
 	for _, frontend := range frontendList.Items {
 		if frontend.Spec.Module != nil {
-			modName := frontend.GetName()
+			// module names in fed-modules.json must be camelCase
+			// K8s does not allow camelCase names, only
+			// whatever-this-case-is, so we convert.
+			modName := strcase.ToLowerCamel(frontend.GetName())
```
Comment (BlakeHolifield) [inline controllers/reconcile.go:410]: We will use our own `ToLowerCamel` function, this was proving our fix for this specific case.

File: hac_nav.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+apiVersion: cloud.redhat.com/v1alpha1
+kind: Bundle
+metadata:
+  name: hac-navigation
+spec:
+  id: hac
+  envName: env-boot
+  title: HAC
+  extraNavItems:
```
Comment (karelhala) [inline hac_nav.yaml:9]: Mhh, we shouldn't be needing this. It should be enough just to add
```yaml
appList:
    - hac-core
```

And change how the navigation is generated in `hac.yml`

File: hac.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    paths:
+      - /apps/hac-core
+  image: quay.io/redhat-cloud-services/hac-core:ci-beta
+  navItems:
+    - appId: "hacCore"
+      title: "Core"
+      href: "/hac"
+    - appId: "hacDev"
+      title: "App Studio"
+      href: "/hac/app-studio"
```
Comment (karelhala) [inline hac.yaml:22]: There should also be a couple of `dynamicNav` NavItems.
```suggestion
  navItems:
    - appId: "hacCore"
      title: "Core"
      href: "/hac"
    - title: "Dynamic nav admin",
      dynamicNav: "hacCore/admin-demo-section"
    - title: "Dynamic nav",
      dynamicNav: "hacCore/dev-demo-section"
    - appId: "hacDev"
      title: "App Studio"
      href: "/hac/app-studio"
```

Comment (BlakeHolifield) [reply hac.yaml]: Ah I see now. Got it all squared away. Thanks!

File: controllers/reconcile.go [hunk 4]
```diff
@@ -369,6 +370,7 @@ func createConfigConfigMap(ctx context.Context, pClient client.Client, frontend
 			bundleCacheMap := make(map[string]crd.BundleNavItem)
 			for _, extraItem := range bundle.Spec.ExtraNavItems {
 				bundleCacheMap[extraItem.Name] = extraItem.NavItem
+				newBundleObject.NavItems = append(newBundleObject.NavItems, extraItem.NavItem)
```
Comment (psav) [inline controllers/reconcile.go:373]: oopsie

File: hac_nav.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+kind: Bundle
+metadata:
+  name: hac-navigation
+spec:
+  id: hac
+  envName: env-boot
+  title: HAC
+  appList:
+  - inventory
+  - hac-core
```
Comment (psav) [inline hac_nav.yaml:11]: newline

File: landing_nav.yaml [hunk 6]
```diff
@@ -6,7 +6,7 @@ spec:
   id: landing-navigation
   envName: env-boot
 customNav:
-  - id: rhel
+  - id: Example
```
Comment (psav) [inline landing_nav.yaml:9]: rhel

**Review by psav — CHANGES_REQUESTED** [review]

**Pull request merged**